test/integration/asyncprofiler_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022-2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/parca-dev/parca-agent/pkg/asyncprofiler"
+	"github.com/parca-dev/parca-agent/pkg/convert"
+)
+
+func TestIntegrationAsyncProfiler(t *testing.T) {
+	cwd, err := os.Getwd()
+	require.NoError(t, err, "Failed to get current working directory")
+	// Start the Java program
+	javaProgramPath := filepath.Join(cwd, "testdata", "java-app", "target", "first-application-0.0.1-SNAPSHOT.jar")
+	cmd := exec.Command("java", "-Xms1G", "-Xmx1G", "-XX:+AlwaysPreTouch", "-jar", javaProgramPath)
+	fmt.Fprintf(os.Stdout, "Executing command: %v\n", cmd)
+	err = cmd.Start()
+	require.NoError(t, err, "Failed to start Java program")
+
+	time.Sleep(50 * time.Second) // We need to wait for the Java program to start
```

Comment:
What is it that we're actually waiting for here? How would we tell from the "production" profiler that it's safe to attach?

Reply:
Oops, sorry. These are leftovers. They were just part of one of the debugging. Same for the ones below. Will remove them.